sequence_python/README.rst

```diff
@@ -0,0 +1,113 @@
+===============================
+Sequence from Python expression
+===============================
+
+.. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fserver--tools-lightgray.png?logo=github
+    :target: https://github.com/OCA/server-tools/tree/13.0/sequence_python
+    :alt: OCA/server-tools
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/server-tools-13-0/server-tools-13-0-sequence_python
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/149/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module allows to generate a sequence by a Python formula expression.
+
+Besides common Python functions and operators, it provides several functions
+like 'random' and 'uuid' in the expression, as well as variables such as the
+next number in the sequence. These are also listed next to the input field
+on the sequence form view.
+
+If you want to add more variables for use in the expression, you can extend
+this module.
+
+Use cases for this module could be:
+
+- You want to generate alphanumeric numbering
+- You want to apply some math to the number to prevent customers from knowing
+  their place in the sequence
+- You want to use UUID
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Go to the form view of an `ir.sequence` record
+* Go to the Python tab
+* Enable the 'Use Python' checkbox
+* Change the default 'number' expression to something more fancy.
+
+Example:
+
+.. code-block:: python
+
+   # To separate the Odoo-generated number with hyphens eg. 0-0-0-0-1
+   '-'.join(number_padded)
+
+   # To have an UUID as the sequence value
+   uuid.uuid4().hex
+
+   # To use an 8-digit binary number
+   {0:#010b}'.format(number)
+
+The possibilities are nearly endless.
```

Comment:
Is it possible to use date and date_range as regular sequence ?
Could you add a note regarding that point.

Reply:
@legalsylvain @thomaspaulb are we talking about prefix and suffix here? AFAICS only the number itself is changed so on a standard 13.0 Bank Journal sequence your next number is previewed as 

> BNK1/2021/57be06c0b96045b09f066e4fbefa59ae

The 8-digit binary number-suggestion nets me `unexpected EOF while parsing (, line 1)` though.

Reply:
The 8-bit example was wrong, I changed it and fixed two other bugs as well.

Reply:
@legalsylvain The date stuff as you can put in the prefix and suffix just works:

![image](https://user-images.githubusercontent.com/1466356/106171894-6acef680-6192-11eb-8039-ef343b33bd7c.png)

![image](https://user-images.githubusercontent.com/1466356/106171970-81754d80-6192-11eb-8329-4e3401837c66.png)

Is that what you mean?

Reply:
Sorry, i did a mistake during the review !
Thanks for this module.

Reply:
@legalsylvain No problem, thanks for the review! Could you flip it to 'approved' ?